kodeco-oop-hw.playground/Contents.swift

```diff
@@ -0,0 +1,214 @@
+// Exercise 1
+class Post {
+    var author: String
+    var content: String
+    var likes = 0
+    
+    init(author: String, content: String) {
+        self.author = author
+        self.content = content
+    }
+    
+    func display() {
+        print("Post from \(author) with \(likes) likes: \(content)")
+    }
+    
+    func incrementLikes() {
+        likes += 1
+    }
+}
+
+// Exercise 2
+class Product {
+    var name: String
+    var price: Double
+    var quantity: Int
+    
+    init(name: String, price: Double, quantity: Int) {
+        self.name = name
+        self.price = price
+        self.quantity = quantity
+    }
+}
+
+class ShoppingCartSingleton {
+    private static var shoppingCartInstance: ShoppingCartSingleton?
+    
+    private var products: [Product] = []
+    
+    private init() {}
+    
+    static func sharedInstance() -> ShoppingCartSingleton {
```

Comment:
Most users of a singleton will expect that the "sharedInstance" is a property rather than a method.